nightfall-optimist/src/services/block-assembler.mjs

```diff
@@ -41,18 +41,18 @@ async function makeBlock(proposer, number = TRANSACTIONS_PER_BLOCK) {
 This function will make a block iff I am the proposer and there are enough
 transactions in the database to assembke a block from. It loops until told to
 stop making blocks. It is called from the 'main()' routine to start it, and
-should not be called from anywhere else because we only want one instance ever.
+should not be called from anywhere else because we only want one instance ever
 */
 export async function conditionalMakeBlock(proposer) {
-  logger.debug('Ready to make blocks');
-  while (makeBlocks) {
-    logger.silly('Block Assembler is waiting for transactions');
-    await nextHigherPriorityQueueHasEmptied(1); // i.e. the highest priority queue is empty
-    // if we are the current proposer, and there are enough transactions waiting
-    // to be processed, we can assemble a block and create a proposal
-    // transaction. If not, we must wait until either we have enough (hooray)
-    // or we're no-longer the proposer (boo).
-    if ((await numberOfUnprocessedTransactions()) >= TRANSACTIONS_PER_BLOCK && proposer.isMe) {
+  // if we are the current proposer, and there are enough transactions waiting
+  // to be processed, we can assemble a block and create a proposal
+  // transaction. If not, we must wait until either we have enough (hooray)
+  // or we're no-longer the proposer (boo).
+  if (proposer.isMe) {
```

Comment:
Is there a reason this is split across two `if` statements?

Reply:
The outer `If` does not re-enqueue `conditonalMakeBlock` since proposer is no longer the host node.